vignettes/rvar.Rmd

```diff
@@ -151,28 +151,63 @@ includes:
 | Array transposition         | `t()`, `aperm()` |
 | Matrix decomposition        | `chol()` |
 
-## Expectations and summaries
+## Expectations and summary functions
 
-The `E()` and `mean()` functions are aliases of each other, producing means
-within each cell of an `rvar`. For example:
+The `E()` function is an alias of `mean()`, producing means
+within each cell of an `rvar`. For example, given `mu`:
 
-```{r E}
-E(mu)
+```{r mu}
+mu
 ```
 
-You can similarly obtain the `median()` or `variance()` of `rvar` cells:
+We can get the expectation of each cell of `mu`:
 
-```{r variance}
-variance(mu)
+```{r E_mu}
+E(mu)
 ```
 
 Expectations of logical expressions are probabilities, and can be computed either
-with `E()` / `mean()` or with `Pr()` (which is purely notational sugar):
+with `E()` / `mean()` or with `Pr()`. `Pr()` is provided as notational sugar, but
+also checks that the input is a logical variable before taking the mean:
 
 ```{r Pr}
 Pr(mu > 0)
 ```
 
+More generally, the `rvar` data type provides two types of summary functions: 
+
+1. Summary functions that mimic base-R vector summary functions, except applied to `rvar` vectors. These apply their summaries **over** elements of the input vectors **within** each draw, generally returning an `rvar` of length 1. These functions are prefixed with `rvar_` as a reminder that they return `rvar`s. Here is an example of `rvar_mean()`:
+  
+    ```{r rvar_mean_mu}
+    rvar_mean(mu)
+    ```
+
+2. Summary functions that summarise **within** elements of input vectors and **over** draws. These summary functions generally return base arrays (`numeric` or `logical`) of the same shape as the input `rvar`, and are especially useful for diagnostic summaries. These summary functions are not prefixed with `rvar_` as they do not return `rvar`s. Here is an example of `mean()`:
+
+    ```{r mean_mu}
```

Comment:
There seem to be a few leading whitespaces here that could be removed.

Reply:
The leading whitespace here ensures that the code block is nested under the list item above when rendered to HTML. It's a bit odd I admit but seems to be the correct way to do that (it must be exactly 4 spaces)